tests/fixture_setup/os_release.py

```diff
@@ -0,0 +1,64 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2018 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from textwrap import dedent
+from unittest import mock
+
+import fixtures
+
+from snapcraft.internal import os_release
+
+
+class FakeOsRelease(fixtures.Fixture):
+
+    def __init__(self, id: str='ubuntu', version_id: str='16.04',
+                 version_codename: str=None) -> None:
+        self._id = id
+        self._version_id = version_id
+        self._version_codename = version_codename
+
+    def _setUp(self):
+        super()._setUp()
+
+        with open('os-release', 'w') as release_file:
+            print(dedent("""\
+                NAME="Ubuntu"
+                VERSION="16.04.3 LTS (Xenial Xerus)"
+                ID_LIKE=debian
+                PRETTY_NAME="Ubuntu 16.04.3 LTS"
+                HOME_URL="http://www.ubuntu.com/"
+                SUPPORT_URL="http://help.ubuntu.com/"
+                BUG_REPORT_URL="http://bugs.launchpad.net/ubuntu/"
+                UBUNTU_CODENAME=xenial"""), file=release_file)
```

Comment:
Is it worth faking everything when you only read some things? I ask because you the ID, VERSION_ID and VERSION_CODENAME might be set through `__init__()` and not correctly go with these items.

Reply:
I leave the others for the other general cases we have, it does indeed not match with what one would generally put into this file though.